### ISSUE TITLE
Add Custom NSG rules to template 

### ISSUE THREAD

**Issue by csubj** [Azure/acs-engine#1870]
**Is this a request for help?**:
No 
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one): 
FEATURE 

---

**What version of acs-engine?**:
ALL 
---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
ALL

**What you expected to happen**:
When the template is used for generating the resource/deployment files a master NSG is created with some default securitygroup rules; as it is there is no current way to override or add custom rules in the template - they need to be added manually after generation. This would be a very nice feature to add for people trying to lock down vnet/subnets for security + compliance when running ACS clusters for production.

**Comment by huydinhle** [comment]
This feature is a very nice to have. Do we have an estimate on when in the future we will look into this? I would love to be able to help because it's one of the features we need at our shop

**Comment by CecileRobertMichon** [comment]
@huydinhle I don't believe anyone is looking at this now, if you'd like to start a PR to enable this we'll be happy to offer guidance/reviews/feedback

**Comment by huydinhle** [comment]
yes, I would love to be assigned to work on this. @CecileRobertMichon. Will start looking into the code but please send any references or help that you can think of. 
Thank you

**Comment by CecileRobertMichon** [comment]
for masters VMAS: https://github.com/Azure/acs-engine/blob/api-versions/parts/k8s/kubernetesmasterresources.t#L87
for masters VMSS: https://github.com/Azure/acs-engine/blob/api-versions/parts/k8s/kubernetesmasterresourcesvmss.t#L80

One way you could do this is modify the template function `GetSecurityRules` that is already being used for DCOS and adapt it for Kubernetes (assuming you're interested in k8s clusters) https://github.com/Azure/acs-engine/blob/api-versions/pkg/acsengine/template_generator.go#L394
